lib/p2p/Pool.ts

```diff
@@ -508,7 +527,9 @@ class Pool extends EventEmitter {
     });
 
     peer.on('error', (err) => {
-      this.logger.error(`peer error (${peer.nodePubKey}): ${err.message}`);
+      if (peer.nodePubKey !== this.handshakeData.nodePubKey) {
```

Comment:
I don't see why `error` event handler will ever be called in case of the reachability check, since `bindPeer` is isn't called.

If I missed something, and it's indeed being called, then I think that this condition (`peer.nodePubKey !== this.handshakeData.nodePubKey`) won't be very clear on the long-run.

We need to mark the peer object somehow. We can add a boolean flag to it, and perhaps another static factory method for `Peer` that will activate it. If it's active, `bindPeer` won't bind. This way no event will ever be triggered, and code will be cleaner. 

I'm also seeing logs from `Peer` (`Socket closed prior to handshake`, `Received data`), but since the latest logger refactoring, these can be easily be eliminated by injecting a dummy logger to `Peer` (we can create it on `Logger` as a static instance).

Reply:
I do like having the check on inbound connections to make sure they're not using our pubkey though. It should never happen, but it doesn't hurt to catch it here either.

Reply:
The inbound connection is calling `bindPeer` I think. And I added the dummy logger but I am not sure if I implemented it the way you meant it. Please take a look. 

These messages are leftover and sent by the inbound connection I think:
```
2018-8-28 15:39:05 [P2P] debug: Connected pre-handshake to peer 127.0.0.1:60270
2018-8-28 15:39:05 [P2P] verbose: Verified reachability of advertised address: localhost:8885
2018-8-28 15:39:05 [P2P] debug: Received data (127.0.0.1:60270): {"header":{"id":"bb36e170-aac7-11e8-b573-fd57249abaf3","hash":"zh51r1Ey5gL3mUTOM1417A==","type":"HELLO"},"body":{"version":"1.0.0-prealpha.1","pairs":["BTC/LTC","ZRX/GNT"],"nodePubKey":"0284643297670606e6f381b8dff161254a8ea1db927eed6c2d33c2015ab4b4dbd6","addresses":[{"host":"localhost","port":8885}]}}ↂ₿ↂ
2018-8-28 15:39:05 [P2P] info: Peer 0284643297670606e6f381b8dff161254a8ea1db927eed6c2d33c2015ab4b4dbd6 socket closed
```